project-base/storefront/components/Blocks/Adverts/Adverts.tsx

```diff
@@ -33,57 +32,57 @@ export const Adverts: FC<AdvertsProps> = ({
     className,
     isSingle,
 }) => {
-    const [{ data: advertsData }] = useAdvertsQuery();
-
-    const filteredAdverts = advertsData?.adverts.filter((advert) => advert.positionName === positionName);
+    const [{ data: advertsData }] = useAdvertsQuery({
+        variables: {
+            categoryUuid: currentCategory?.uuid || null,
+            positionName,
+        },
+    });
```

Comment:
I have just realized that after changing these params, SSR fetching will not work (can be seen by the fact that this query is fetched on the client). To fix this, you have to make sure that `{ query: AdvertsQueryDocument }` in `initServerSideProps.ts` is called with the same params as here.